selenium_tests/manage_people/remove_users_tests.py

```diff
@@ -48,19 +48,24 @@ def test_remove_users(self, test_case_id,
 
         user_list_page = UserListPageObject(self.driver)
 
-        # note delete_user() requires the Canvas role name, not our display
-        # name (e.g. ObserverEnrollment, not Observer)
-
         # Note: the refresh has been added due to issues with remove.  If
         # user is added via the api but the page isn't reload, the tests will
         # sporadically fail out with NoSuchElementException.
         self.driver.refresh()
         self.setUp()
 
+        # Delete the user in Canvas
         user_list_page.delete_user(test_univ_id, canvas_role)
 
-        # Verify that user is not on the page after getting removed
         self.assertTrue(user_list_page.is_loaded())
+        # Note: the refresh has been added due to issues with remove. If the
+        # user has been deleted via the UI, but the page isn't reloaded,
+        # it would sporadically fail out with an assertion error.  We have
+        # yet to find a more graceful solution.
+        self.driver.refresh()
+        self.setUp()
```

Comment:
Yes.  I hesitated to do it that way. :\  but it works and it works very reliably.  Otherwise, it sporadically fails.